src/MergeContext.js

```diff
@@ -1063,14 +1264,39 @@ class PullRequest {
         const waitForMergeable = !(this._stagedPosition && this._stagedPosition.merged());
         const pr = await GH.getPR(this._prNumber(), waitForMergeable);
         assert(pr.number === this._prNumber());
+
         this._rawPr = pr;
+
+        if (this._prMergeable()) {
+            const mergeSha = await GH.getReference("pull/" + this._prNumber() + "/merge");
+            this._mergeCommit = await GH.getCommit(mergeSha);
+            try {
+                this._commitMessage = new CommitMessage(this._rawPr, this._mergeCommit.author);
+            } catch (e) {
+                this._logEx(e, "cannot parse commit message");
+            }
+        }
     }
 
-    // Whether the commit message configuration remained intact since staging.
-    async _messageIsFresh() {
-        const result = this._prMessage() === this._stagedCommit.message;
+    // Whether the staged commit metadata remained intact since staging.
+    _stagedCommitMetadataIsFresh() {
+        if (!this._commitMessage) {
+            this._log("staged commit message became invalid (and will be treated as stale)");
+            return false;
+        }
+        const result = this._commitMessage.whole() === this._stagedCommit.message;
         this._log("staged commit message freshness: " + result);
-        return result;
+        if (!result)
+            return false;
+
+        const oldAuthor = this._stagedCommit.author;
+        const newAuthor = this._commitMessage.author();
+        const authorIsFresh = oldAuthor.name === newAuthor.name && oldAuthor.email === newAuthor.email;
+        this._log("staged commit author freshness: " + authorIsFresh);
```

Comment:
Why are we ignoring (i.e. not comparing) the author date here? The _commitMessage.author() gets its date from this._mergeCommit and here we are examining this._stagedCommit so it feels wrong to assume the two are the same. Or perhaps we are assuming that they may change and that change should be ignored for some reason? In that case, we should add a source code comment to explain.

Reply:
We compare dates in the outer _stagedCommitIsFresh() method, just after comparing tree SHAs. I moved these checks here to have all of them in one place (c486b1). Also I the old check compared commiter.date, but it should compare author.date instead.